Tools/visualize_log.py

```diff
@@ -1,221 +1,211 @@
-        downsampled = agent_data.iloc[::10]
-
-        ax.plot(downsampled["AgentX"],
-                downsampled["AgentZ"],
-                downsampled["AgentY"],
-                color=color,
-                alpha=0.5,
-                linewidth=0.5,
-                label=f"Agent Type: {agent_type}")
-        agent_types.add(agent_type)
-
-    # Define event markers with higher zorder for visibility.
+
+        ax.plot(
+            agent_data["AgentX"],
+            agent_data["AgentZ"],
+            agent_data["AgentY"],
+            color=color,
+            alpha=0.5,
+            linewidth=0.5,
+        )
+
+    # Define event markers for different events.
     event_markers = {
-        "HIT": ("o", "green", "Hit"),
-        "MISS": ("x", "red", "Miss"),
-        "NEW_THREAT": ("^", "orange", "New Threat"),
-        "NEW_INTERCEPTOR": ("s", "blue", "New Interceptor")
+        "INTERCEPTOR_HIT": ("o", "lime", "Hit"),
+        "INTERCEPTOR_MISS": ("X", "red", "Miss"),
+        "NEW_THREAT": ("^", "darkorange", "New Threat"),
+        "NEW_INTERCEPTOR": ("s", "blue", "New Interceptor"),
     }
 
-    # Plot events.
+    # Plot the events.
     for event_type, (marker, color, label) in event_markers.items():
         event_data = event_df[event_df["Event"] == event_type]
         if not event_data.empty:
             ax.scatter(
                 event_data["PositionX"],
                 event_data["PositionZ"],
                 event_data["PositionY"],
+                s=60,
                 c=color,
                 marker=marker,
-                s=100,  # Increased marker size for better visibility.
-                label=label,
-                edgecolors="k",  # Adding black edges for contrast.
                 depthshade=True,
-                zorder=5  # Ensure markers are on top.
+                edgecolors="black",
+                zorder=5,
+                label=label,
             )
 
-    # Set labels.
-    ax.set_xlabel("X (m)", fontsize=12)
-    ax.set_ylabel("Z (m)", fontsize=12)
-    ax.set_zlabel("Y (m)", fontsize=12)
-
-    # Set view angle for better visualization.
-    ax.view_init(elev=20, azim=45)
-
     # Add a ground plane for reference.
     x_min, x_max = ax.get_xlim()
     z_min, z_max = ax.get_ylim()
-    xx, zz = np.meshgrid(np.linspace(x_min, x_max, 2),
-                         np.linspace(z_min, z_max, 2))
+    xx, zz = np.meshgrid([x_min, x_max], [z_min, z_max])
     yy = np.zeros_like(xx)
-    ax.plot_surface(xx, zz, yy, alpha=0.2, color="green")
-
-    plt.title("Agents Trajectories and Events (X: Right, Z: Forward, Y: Up)",
-              fontsize=14)
-
-    # Optimize legend to prevent overcrowding.
-    handles, labels = ax.get_legend_handles_labels()
-    # Remove duplicate labels.
-    unique = dict(zip(labels, handles))
-    ax.legend(unique.values(),
-              unique.keys(),
-              loc="upper left",
-              bbox_to_anchor=(1, 1),
-              fontsize=10)
-
```

Comment:
No more legend :(

Reply:
Fixed.  Looks like this now:

<img width="859" height="630" alt="Image" src="https://github.com/user-attachments/assets/90557524-da7e-4e60-ba5f-a0c4373a0d8c" />